src/vs/workbench/contrib/chat/browser/chatContentParts/chatMultiDiffContentPart.ts

```diff
@@ -143,19 +143,17 @@ export class ChatMultiDiffContentPart extends Disposable implements IChatContent
 		}));
 		const setupActionBar = () => {
 			actionBar.clear();
-
+			const type = getChatSessionType(this._element.sessionResource);
 			let marshalledUri: unknown | undefined = undefined;
 			let contextKeyService: IContextKeyService = this.contextKeyService;
-			if (this.editorService.activeEditor instanceof ChatEditorInput) {
-				contextKeyService = this.contextKeyService.createOverlay([
-					[ChatContextKeys.agentSessionType.key, this.editorService.activeEditor.getSessionType()]
-				]);
-
-				marshalledUri = {
-					...this.editorService.activeEditor.resource,
-					$mid: MarshalledId.Uri
-				};
-			}
+
+			contextKeyService = this.contextKeyService.createOverlay([
+				[ChatContextKeys.sessionType.key, type]
+			]);
+			marshalledUri = {
+				...this._element.sessionResource,
+				$mid: MarshalledId.Uri
```

Comment:
I know this was there before but this seems strange. @osortega during debt week can you please see why we need to manually marshal the uri here